databus-connector/src/main/java/com/sequenceiq/cloudbreak/sigmadbus/SigmaDatabusClient.java

```diff
@@ -0,0 +1,160 @@
+        this.sigmaDatabusConfig = sigmaDatabusConfig;
+        this.databusStreamConfiguration = databusStreamConfiguration;
+    }
+
+    /**
+     * Upload data into databus. If the payload is larger than 1 MB, the data will be uploaded to cloudera S3.
+     * @param request databus record payload input
+     * @throws DatabusRecordProcessingException error during databus record processing
+     */
+    public void putRecord(DatabusRequest request) throws DatabusRecordProcessingException {
+        ManagedChannelWrapper channelWrapper = getMessageWrapper();
+        DbusProto.PutRecordRequest recordRequest = convert(request, databusStreamConfiguration);
+        String requestId =  RequestIdUtil.newRequestId();
+        LOGGER.debug("Creating databus request with request id: {}", requestId);
+        buildMdcContext(request, requestId);
+        DbusProto.PutRecordResponse recordResponse = newStub(channelWrapper.getChannel(),
+                requestId, ThreadBasedUserCrnProvider.INTERNAL_ACTOR_CRN)
+                .putRecord(recordRequest);
+        DbusProto.Record.Reply.Status status = recordResponse.getRecord().getStatus();
+        if (DbusProto.Record.Reply.Status.SENT.equals(status)) {
+            String recordId = recordResponse.getRecord().getRecordId();
+            LOGGER.debug("Dbus record sucessfully processed with record id: {}", recordId);
+        } else if (DbusProto.Record.Reply.Status.PENDING.equals(status)) {
+            String recordId = recordResponse.getRecord().getRecordId();
+            String s3BucketUrl = recordResponse.getRecord().getUploadUrl();
+            LOGGER.debug("Dbus record can be uploaded to s3 [record id: {}], [s3 url: {}]", recordId, s3BucketUrl);
+            uploadRecordToS3(s3BucketUrl, request, recordId);
+        } else {
+            throw new DatabusRecordProcessingException("Cannot process record to Sigma Databus.");
+        }
+    }
+
+    private void uploadRecordToS3(String s3Url, DatabusRequest request, String recordId) throws DatabusRecordProcessingException {
+            String payload = getPayload(request);
+        try {
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest httpRequest = HttpRequest.newBuilder()
+                    .uri(new URI(s3Url))
+                    .PUT(HttpRequest.BodyPublishers.ofString(payload))
+                    .build();
+            HttpResponse<String> response = client.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            Response.Status.Family statusFamily = Response.Status.Family.familyOf(response.statusCode());
+            if (Response.Status.Family.SUCCESSFUL.equals(statusFamily) || Response.Status.Family.REDIRECTION.equals(statusFamily)) {
+                LOGGER.debug("Databus record with id {} successfully uploaded to s3.", recordId);
+            } else {
+                throw new DatabusRecordProcessingException(String.format("S3 upload failed for databus record with id %s", recordId));
```

Comment:
it would be good to emit a LOGGER.warn with the Response Status and maybe reason as a clue why it's failing.

Reply:
yeah, could be, i choosed to just logging out the status code itself. before that exception, which is useful for the success operations as well